pkg/network/tracer/tracer.go

```diff
@@ -251,46 +251,55 @@ func (t *Tracer) start() error {
 	return nil
 }
 
+func loadEbpfConntracker(cfg *config.Config, telemetryComponent telemetryComponent.Component) (netlink.Conntracker, error) {
+	if !cfg.EnableEbpfConntracker {
+		log.Info("ebpf conntracker disabled")
+		return nil, nil
+	}
+
+	if err := netlink.LoadNfConntrackKernelModule(cfg); err != nil {
+		log.Warnf("failed to load conntrack kernel module, though it may already be loaded: %s", err)
+	}
+
+	return NewEBPFConntracker(cfg, telemetryComponent)
+}
+
 func newConntracker(cfg *config.Config, telemetryComponent telemetryComponent.Component) (netlink.Conntracker, error) {
 	if !cfg.EnableConntrack {
 		return netlink.NewNoOpConntracker(), nil
 	}
 
+	var clb netlink.Conntracker
 	var c netlink.Conntracker
 	var err error
-
 	if !cfg.EnableEbpfless {
-		ns, err := cfg.GetRootNetNs()
-		if err != nil {
-			log.Warnf("error fetching root net namespace, will not attempt to load nf_conntrack_netlink module: %s", err)
-		} else {
-			defer ns.Close()
-			if err = netlink.LoadNfConntrackKernelModule(ns); err != nil {
-				log.Warnf("failed to load conntrack kernel module, though it may already be loaded: %s", err)
-			}
-		}
-		if cfg.EnableEbpfConntracker {
-			if c, err = NewEBPFConntracker(cfg, telemetryComponent); err == nil {
-				return c, nil
-			}
+		if c, err = loadEbpfConntracker(cfg, telemetryComponent); err != nil {
 			log.Warnf("error initializing ebpf conntracker: %s", err)
-		} else {
-			log.Info("ebpf conntracker disabled")
+			log.Info("falling back to netlink conntracker")
 		}
 
-		log.Info("falling back to netlink conntracker")
+		if clb, err = newCiliumLoadBalancerConntracker(cfg); err != nil {
+			log.Warnf("cilium lb conntracker is enabled, but failed to load: %s", err)
+		}
 	}
 
-	if c, err = netlink.NewConntracker(cfg, telemetryComponent); err == nil {
-		return c, nil
+	if c == nil {
+		if c, err = netlink.NewConntracker(cfg, telemetryComponent); err != nil {
+			if errors.Is(err, netlink.ErrNotPermitted) || cfg.IgnoreConntrackInitFailure {
+				log.Warnf("could not initialize netlink conntracker: %s", err)
+			} else {
+				return nil, fmt.Errorf("error initializing conntracker: %s. set network_config.ignore_conntrack_init_failure to true to ignore conntrack failures on startup", err)
+			}
+		}
 	}
 
-	if errors.Is(err, netlink.ErrNotPermitted) || cfg.IgnoreConntrackInitFailure {
-		log.Warnf("could not initialize conntrack, tracer will continue without NAT tracking: %s", err)
-		return netlink.NewNoOpConntracker(), nil
+	c = chainConntrackers(c, clb)
+	if c.GetType() == "" {
```

Comment:
is there a way to read the impl type directly here?

Reply:
We can, be we already have `GetType()` as part of the interface, so why not use it here.

Reply:
We can, be we already have `GetType()` as part of the interface, so why not use it here.

Reply:
We can, be we already have `GetType()` as part of the interface, so why not use it here.